lib/providers/profiles_provider.dart

```diff
@@ -149,22 +148,39 @@ class ProfilesProvider with ChangeNotifier {
       if (response.statusCode < 400) {
         var decodedBody = json.decode(response.body);
         List<Profile> fetchedProfiles = [];
+        List<Profile> sortedProfiles = [];
         List<Transaction> fetchedTransactions = [];
 
         if (decodedBody is List) {
-          for (var i = 0, j = 0; i < decodedBody.length; i++, j++) {
-            var element = decodedBody[i];
+          for (var element in decodedBody) {
             var profileGuid = element["guid"];
 
-            var profileTransactions = await _fetchTransactions(profileGuid);
+            fetchedProfiles.add(
+              Profile(
+                guid: profileGuid,
+                name: element["name"],
+                balance: element["balance"],
+                monster: Monsters.blue,
+                createdAt: element["createdAt"],
+              ),
+            );
+          }
+          fetchedProfiles.sort();
+
+          for (var i = 0, j = 0; i < fetchedProfiles.length; i++, j++) {
+            var profile = fetchedProfiles[i];
+
+            var profileTransactions = await _fetchTransactions(profile.guid);
             fetchedTransactions.addAll(profileTransactions);
 
-            fetchedProfiles.add(
+            sortedProfiles.add(
               Profile(
-                  guid: profileGuid,
-                  name: element["name"],
-                  balance: element["balance"],
-                  monster: Monsters.values[j]),
+                guid: profile.guid,
+                name: profile.name,
+                balance: profile.balance,
+                monster: Monsters.values[j],
+                createdAt: profile.createdAt,
+              ),
```

Comment:
This as a whole could be replace by a copyWith method (typical dart method of copying objects with different values)

```
Child copyWith({
    String? guid,
    String? name,
    double? balance,
    String? createdAt,
    Monsters? monster,
  }) {
    return Child(
      guid: name ?? this.name,
      name: name ?? this.name,
      balance: balance ?? this.balance,
      createdAt: createdAt ?? this.createdAt,
      monster: monster ?? this.monster,
    );
  }
```
(This would have been part of the generated code with freezed haha)